### PR TITLE
New tutorial: snap CD from travis

### PR DESCRIPTION
**Pull request by caldav**
Markdown port and review for #239

### CODE REVIEW

**Comment by caldav** [comment]
@elopio, small updates here and there, please check

File: tutorials/continuous-snap-delivery-from-travis-ci/continuous-snap-delivery-from-travis-ci.md [hunk 1]
```diff
@@ -0,0 +1,364 @@
...
+```
+
+Later you can use Travis CI to run your tests. For that you will have to adjust this file depending on the language of your project, and add the install and script steps. But for now, we don’t need any of that as we’ll just make continuous delivery to the store.
+
+In the next step we’ll make sure that your snap can be built successfully in Travis CI, so we can push the generated snap to the store.
+
+## Building the snap in a clean environment
+Duration: 5:00
+
+Travis CI executors are using Ubuntu 14.04, but are missing some kernel changes to be able to run snapcraft as a snap itself. Since they support Docker, we will use a snapcraft Docker container to build the snap.
```
Comment (evilnick) [inline tutorials/continuous-snap-delivery-from-travis-ci/continuous-snap-delivery-from-travis-ci.md]: I would say:

Travis CI uses a build environment based on Ubuntu 14.04, which is missing some requirements for running snapcraft. However, it does have support for Docker, so we will use a Docker container to build the snap

File: tutorials/continuous-snap-delivery-from-travis-ci/continuous-snap-delivery-from-travis-ci.md [hunk 2]
```diff
@@ -0,0 +1,364 @@
...
+
+Before we build our snap in Travis CI’s executors, we need to make sure the snap builds correctly in a Docker container.
+
+Positive
+: **The need for a clean environment**
+When you build a snap on your machine, it could be using some dependencies that are already installed but you forgot to declare as `build-packages` in the snapcraft.yaml.
+This means that when somebody will try to build the same snap on a different machine, those undeclared packages could be missing and the build would fail.
```
Comment (evilnick) [inline tutorials/continuous-snap-delivery-from-travis-ci/continuous-snap-delivery-from-travis-ci.md]: s /will try/tries/

File: tutorials/continuous-snap-delivery-from-travis-ci/continuous-snap-delivery-from-travis-ci.md [hunk 3]
```diff
@@ -0,0 +1,364 @@
...
+A clean environment, such as a default Ubuntu 14.04 or 16.04 in a container, ensures reproducible builds.
+
+Let’s install docker, to get the same environment that Travis CI will use to build the snap:
```
Comment (evilnick) [inline tutorials/continuous-snap-delivery-from-travis-ci/continuous-snap-delivery-from-travis-ci.md]: Docker?

File: tutorials/continuous-snap-delivery-from-travis-ci/continuous-snap-delivery-from-travis-ci.md [hunk 4]
```diff
@@ -0,0 +1,364 @@
...
+Once you get the snap building in the Docker container, you can be confident that it will build without problems in Travis CI.
+
+We are ready to enable continuous delivery!
+
+## Building and pushing the snap in Travis CI
+Duration: 6:00
+
+The final step to enable continuous delivery from Travis CI to all your early adopters is to add some commands to the `.travis.yml` file we created before. But we don’t really have to do much, snapcraft will take care of it for us.
+
+We need install the travis command-line utility, which will be used by snapcraft in a moment. Travis is a ruby gem, but is not available in the Ubuntu repositories, so installing it requires a few commands:
```
Comment (evilnick) [inline tutorials/continuous-snap-delivery-from-travis-ci/continuous-snap-delivery-from-travis-ci.md]: need **to** install...

File: tutorials/continuous-snap-delivery-from-travis-ci/continuous-snap-delivery-from-travis-ci.md [hunk 5]
```diff
@@ -0,0 +1,364 @@
...
+In the next step we’ll make sure that your snap can be built successfully in Travis CI, so we can push the generated snap to the store.
+
+## Building the snap in a clean environment
+Duration: 5:00
+
+Travis CI uses a build environment based on Ubuntu 14.04, which is missing some requirements for running snapcraft. However, it does have support for Docker, so we will use a Docker container to build the snap.
+
+Before we build our snap in Travis CI’s executors, we need to make sure the snap builds correctly in a Docker container.
+
+Positive
```
Comment (elopio) [inline tutorials/continuous-snap-delivery-from-travis-ci/continuous-snap-delivery-from-travis-ci.md]: What does this positive mean?

Comment (caldav) [inline tutorials/continuous-snap-delivery-from-travis-ci/continuous-snap-delivery-from-travis-ci.md]: It creates an orange info box ("negative" creates a red one)

Comment (caldav) [inline tutorials/continuous-snap-delivery-from-travis-ci/continuous-snap-delivery-from-travis-ci.md]: screenshot from 2017-07-13 19-08-46

Comment (elopio) [inline tutorials/continuous-snap-delivery-from-travis-ci/continuous-snap-delivery-from-travis-ci.md]: Nice, thanks for the info. I suggest to make the name more clear, calling it like `positive note`. But I will sure enjoy a lot more writing tutorials now in markdown :D

**Comment by evilnick** [comment]
@elopio We look forward to seeing more - the Markdown we use is still a bit ...quirky, but 1000 times better than having to use a google doc!

**Pull request merged**